### PR TITLE
Add benchmarks for the geosearch

### LINKED ISSUE #336: Add GeoSearch test to benchmarks (for indexation and search)
Add tests for geosearch in the current benchmarks
- [ ] for indexation
- [ ] for search

### PR DESCRIPTION
**Pull request by irevoire**
closes #336

Should I merge this PR in #322 and then we merge everything in `main` or should we wait for #322 to be merged and then merge this one in `main` later?

### CODE REVIEW

**Comment by irevoire** [comment]
Oops I forgot to update the README and the `bench.yaml`. I'm putting this in draft PR for now sorry

File: benchmarks/README.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ To run all the benchmarks (~5h):
 cargo bench
 ```
 
-To run only the `songs` (~1h), `wiki` (~3h) or `indexing` (~4h) benchmark:
+To run only the `search songs` (~1h), `search_wiki` (~3h), `search_geo` (~20m) or `indexing` (~2h) benchmark:
```
Comment (curquiza) [inline benchmarks/README.md:39]:
```suggestion
To run only the `search_songs` (~1h), `search_wiki` (~3h), `search_geo` (~20m) or `indexing` (~2h) benchmark:
```

File: benchmarks/README.md [hunk 2]
```diff
@@ -81,36 +81,52 @@ Run the comparison script:
...
-`wiki` is a subset of the [`wikipedia-articles.csv` dataset](https://milli-benchmarks.fra1.digitaloceanspaces.com/datasets/wiki-articles.csv.gz).
+`smol-wiki` is a subset of the [`wikipedia-articles.csv` dataset](https://milli-benchmarks.fra1.digitaloceanspaces.com/datasets/wiki-articles.csv.gz).
 
 It was generated with the following command:
 
 ```bash
 xsv sample --seed 42 500000 wiki-articles.csv -o smol-wiki-articles.csv
 ```
 
+_[Download the `smol-wiki` dataset](https://milli-benchmarks.fra1.digitaloceanspaces.com/datasets/smol-wiki.csv.gz)._
```
Comment (curquiza) [inline benchmarks/README.md:112]:
```suggestion
_[Download the `smol-wiki` dataset](https://milli-benchmarks.fra1.digitaloceanspaces.com/datasets/smol-wiki-articles.csv.gz)._
```

File: benchmarks/README.md [hunk 3]
```diff
@@ -81,36 +81,52 @@ Run the comparison script:
...
 ### Movies
 
 `movies` is a really small dataset we uses as our example in the [getting started](https://docs.meilisearch.com/learn/getting_started/)
 
 _[Download the `movies` dataset](https://docs.meilisearch.com/movies.json)._
 
+
+### All Countries
+
+`smol-all-countries` is a subset of the [`all-countries.csv` dataset]()
```
Comment (curquiza) [inline benchmarks/README.md:123]:
```suggestion
`smol-all-countries` is a subset of the [`all-countries.csv` dataset](https://milli-benchmarks.fra1.digitaloceanspaces.com/datasets/all-countries.csv.gz)
```

**Review by curquiza — CHANGES_REQUESTED** [review]

**Review by curquiza — APPROVED** [review]
Thank you Tamo!!

**Comment by irevoire** [comment]
@Kerollmops, should I merge this in the #322, or should we first merge the geosearch and then the benchmarks on main? What do you prefer?

**Comment by Kerollmops** [comment]
I prefer that we merge #322 and this one after :)

**Comment by irevoire** [comment]
Ok so I guess we can merge now :grin:

**Review by Kerollmops — APPROVED** [review]
Thank you!
Have you tested it with the CI, already?

**Comment by irevoire** [comment]
Yes!
But now I'm wondering, maybe we could add a geosearch on a really small dataset? ~10.000 documents

**Comment by Kerollmops** [comment]
How much time does it take to run this one? If it is much longer than the other ones, maybe we should indeed!

**Comment by irevoire** [comment]
No, currently, all the bench related to the geosearch are super fast. It takes 20 minutes for the search and an insignificant amount of time in the indexing benchmark (1h50m5s before the geosearch vs 1h50m6s with this branch).

So I think it would be good to keep this dataset of 1M documents because it's what we want to improve.
But at the same time, add a really small dataset to ensure we don't lose any performance on small datasets.

**Review by Kerollmops — APPROVED** [review]
You can open a new PR to add a small dataset if you want, I prefer to merge this one for now :)
bors merge

**Pull request merged**